alg/gdaltransformer.cpp

```diff
@@ -527,18 +534,18 @@ GDALSuggestedWarpOutput2( GDALDatasetH hSrcDS,
     constexpr int SIGN_FINAL_UNINIT = -2;
     constexpr int SIGN_FINAL_INVALID = 0;
     int iSignDiscontinuity = SIGN_FINAL_UNINIT;
+    int nFailedCount = 0;
+    int iSignArray[2] = {-1, 1};
     for( int i = 0; i < nSamplePoints; i++ )
     {
-        if( !pabSuccess[i] )
-            nFailedCount++;
-        else
+        if( pabSuccess[i] )
         {
             // Fix for https://trac.osgeo.org/gdal/ticket/7243
             // where echo "-2050000.000 2050000.000" |
             //              gdaltransform -s_srs EPSG:3411 -t_srs EPSG:4326
             // gives "-180 63.691332898492"
             // but we would rather like 180
-            if( iSignDiscontinuity == 1 || iSignDiscontinuity == -1 )
+            if( iSignDiscontinuity&0x01 )
```

Comment:
too smart change, and relies on the complement-to-two representation of signed integers